apps/src/templates/SchoolDataInputs.jsx

```diff
@@ -1,99 +1,138 @@
+import React, {useState, useEffect} from 'react';
 import PropTypes from 'prop-types';
-import React from 'react';
 import i18n from '@cdo/locale';
-import {Heading2, BodyTwoText} from '@cdo/apps/componentLibrary/typography';
+import {
+  Heading2,
+  BodyTwoText,
+  BodyThreeText,
+} from '@cdo/apps/componentLibrary/typography';
 import style from './school-association.module.scss';
 import SimpleDropdown from '../componentLibrary/simpleDropdown/SimpleDropdown';
 import {COUNTRIES} from '@cdo/apps/geographyConstants';
 
-const TEST_SCHOOL_DATA = [{value: 'test', text: 'VeryLongNameTestSchool'}];
-export default function SchoolDataInputs({
-  onCountryChange,
-  onSchoolChange,
-  onSchoolNotFoundChange,
-  country = '',
-  schoolType = '',
-  ncesSchoolId = '',
-  schoolName = '',
-  schoolCity = '',
-  schoolState = '',
-  schoolZip = '',
-  schoolLocation = '',
-  useLocationSearch = false,
+export default function SchoolDataInputs(
+  includeHeaders = true,
   fieldNames = {
-    schoolType: 'user[school_info_attributes][school_type]',
     country: 'user[school_info_attributes][country]',
     ncesSchoolId: 'user[school_info_attributes][school_id]',
     schoolName: 'user[school_info_attributes][school_name]',
-    schoolState: 'user[school_info_attributes][school_state]',
     schoolZip: 'user[school_info_attributes][school_zip]',
-    googleLocation: 'user[school_info_attributes][full_address]',
-  },
-  showErrors,
-  showRequiredIndicator,
-}) {
-  let COUNTRY_ITEMS = [];
+  }
+) {
+  const [askForZip, setAskForZip] = useState(false);
+  const [isOutsideUS, setIsOutsideUS] = useState(false);
+  const [zip, setZip] = useState('');
+  const [country, setCountry] = useState('');
+  const [schoolName, setSchoolName] = useState('');
+  const [zipSearchReady, setZipSearchReady] = useState(false);
+
+  let COUNTRY_ITEMS = [{value: 'selectCountry', text: i18n.selectCountry()}];
   for (const item of Object.values(COUNTRIES)) {
     COUNTRY_ITEMS.push({value: item.label, text: item.value});
   }
 
+  useEffect(() => {
+    if (zip.length === 5) {
+      setZipSearchReady(true);
+    } else {
+      // Removes the school dropdown if you delete part of the zip
+      setZipSearchReady(false);
+    }
+  }, [zip.length]);
+
+  const onCountryChange = e => {
+    const country = e.target.value;
+    setCountry(country);
+    // We don't want to display any fields to start that won't eventually be
+    // necessary, so updating both of these any time country changes
+    if (country === 'US') {
+      setAskForZip(true);
+      setIsOutsideUS(false);
+    } else {
+      setAskForZip(false);
+      setIsOutsideUS(true);
+    }
+  };
+
   return (
     <div className={style.outerContainer}>
-      <Heading2 className={style.topPadding}>{i18n.censusHeading()}</Heading2>
-      <BodyTwoText>{i18n.schoolInfoInterstitialTitle()}</BodyTwoText>
+      {includeHeaders && (
```

Comment:
I can imagine a future where we're looking to grab the school association fields without the big headers at the top

Reply:
Is that in the spec? I don't want to prematurely optimize for a case that doesn't exist yet (or won't soon)

Reply:
We use this component in other places such as the teacher application, afe, yourschool, etc. Moshe is supposed to be look at how the component should work in those other locations but we haven't finalized that yet.

Reply:
@bethanyaconnor because Dani and I had to address this already with the different applications in the areas that I would've expected to include headers, I thought it was a good idea (it was also a small change, so didn't take me much time to build in). I'm trying to design something that is somewhat flexible from the start (especially considering what we talked over yesterday about including this work in places like account settings as phase one of sign up work). But happy to remove it if we don't think we'll need it!